src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/RenameGroupForm.tsx

```diff
@@ -0,0 +1,172 @@
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      groupName: currentGroupName,
+    },
+  });
+
+  /**
+   * The type of the form inputs based on the schema
+   */
+  type Inputs = z.infer<typeof formSchema>;
+
+  /**
+   * Process the form submission.
+   * @param values - The form values
+   */
+  const processSubmit: SubmitHandler<Inputs> = async (values) => {
+    const { groupName } = values;
+    /**Replace console.log with rename api method in here, after it's implemented */
+    alert(`Rename to ${groupName}`);
+    form.reset();
+    onClose();
+  };
+
+  /**
+   * Handle the escape key down event.
+   * @param event - The keyboard event
+   */
+  const handleEscKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      onClose();
+    }
+  };
+
+  /**
+   * When the input is blurred,
+   * process the form submission and close the form.
+   */
+  const handleInputBlur = () => {
+    // if the input is blurred because of the cross button click, don't process the form submission.
+    if (!shouldHandleInputBlur.current) return;
+    // process the form submission
+    processSubmit(form.getValues());
+    onClose();
+  };
+
+  /**
+   * When cross button is mouse downed,
+   * set the shouldHandleBlur to false,
+   * and empty the input value.
+   */
+  const handleCrossButtonMouseDown = () => {
+    shouldHandleInputBlur.current = false;
+    form.setValue('groupName', '');
+  };
+
+  /**
+   * When cross button click is completed,
+   * set the focus back to the input,
+   * and set the shouldHandleBlur to true.
+   */
+  const handleCrossButtonClick = () => {
+    inputRef.current?.focus();
+    shouldHandleInputBlur.current = true;
+  };
+
+  /**
+   * Reset the form when the form is closed.
+   */
+  useEffect(() => {
+    if (!isOpen) form.reset();
+  }, [isOpen, form]);
+
+  /**
+   * Focus the input when the form is opened.
+   */
+  useEffect(() => {
+    if (isOpen) {
+      /**
+       * Needed 350 ms delay to focus the input
+       * because the focus will be set to the radix dropdownMenu trigger on closing the dropdownMenu.
+       * Radix dropdownMenu trigger focus will happen around 300 ms after the dropdownMenu is closed.
+       */
+      setTimeout(() => {
+        inputRef.current?.focus();
+      }, 350);
+    }
+  }, [isOpen]);
+
+  return (
+    <div className="flex grow flex-col gap-3 pl-4 py-2">
+      <Form {...form}>
+        <form onSubmit={form.handleSubmit(processSubmit)}>
+          <FormField
+            control={form.control}
+            name="groupName"
+            render={({ field }) => (
+              <FormItem>
+                <FormControl>
+                  <div className="relative flex items-center">
+                    <Input
+                      type="text"
+                      variant="square"
+                      className={cn(
+                        'text-lg pr-10',
```

Comment:
This is for preventing the text overlapping the cross button to happen.
https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/4c89f4ac-60be-4d74-8acd-2deeed5400c3

Set it `10` because the width of cross button is `w-10`!